packages/solana/src/connectors/walletAdapters.ts

```diff
@@ -0,0 +1,39 @@
+import { PhantomWalletAdapter } from '@solana/wallet-adapter-phantom'
+import { SolflareWalletAdapter } from '@solana/wallet-adapter-solflare'
+import { TrustWalletAdapter } from '@solana/wallet-adapter-trust'
+import { BackpackWalletAdapter } from '@solana/wallet-adapter-backpack'
+
+import type { BaseWalletAdapter } from '@solana/wallet-adapter-base'
+import type { Connector } from '@web3modal/scaffold'
+
+export type AdapterKey = 'phantom' | 'solflare' | 'trustWallet' | 'backpack'
+export const supportedWallets: AdapterKey[] = ['phantom', 'solflare', 'trustWallet', 'backpack']
+
+export function createWalletAdapters() {
+  return {
+    phantom: new PhantomWalletAdapter(),
+    trustWallet: new TrustWalletAdapter(),
+    backpack: new BackpackWalletAdapter(),
+    solflare: new SolflareWalletAdapter()
+  }
+}
+
+export function syncInjectedWallets(
+  w3mConnectors: Connector[],
+  adapters: Record<AdapterKey, BaseWalletAdapter>
+) {
+  supportedWallets.forEach(wallet => {
+    if (window[wallet as keyof Window]) {
+      w3mConnectors.push({
+        id: adapters[wallet].name,
+        type: 'ANNOUNCED',
+        imageUrl: adapters[wallet].icon,
+        name: adapters[wallet].name,
+        provider: adapters[wallet],
+        info: {
+          rdns: `app.${wallet}`
```

Comment:
Template literals for constructing values like `app.${wallet}` dynamically can raise concerns, especially in contexts where the input (in this case, `wallet`) is user-controlled or not strictly validated.

Template literals do not inherently validate the input. Using property access (`adapters[wallet].rdns`) ensures that only predefined keys in the adapters object can be accessed.

Direct property access also makes the code more readable and easier to understand. It's clear that you are accessing a specific property of an object, which makes the codebase easier to maintain and debug.

```suggestion
          rdns: adapters[wallet].rdns
```